Source/VCGeneration/Checker.cs

```diff
@@ -61,6 +61,12 @@ public void GetReady()
     public void GoBackToIdle()
     {
       Contract.Requires(IsBusy);
+      if (Options.ModelViewFile != null) {
+        // Don't re-use theorem provers whose ProverContext still needs to be queried to extract model data.
+        Pool.CheckerDied();
+        Close();
+        return;
+      }
```

Comment:
It seems slightly unfortunate to have what seems like a functional change among what is mostly refactoring, though it seems like a good change.

Reply:
I've added this bugfix in this PR because I'm testing this Boogie PR against Dafny master and the current Boogie master does not work with Dafny without this fix.

I will add a test in Boogie for the bug this fix is for as well but I didn't get to that yet.